samtranslator/model/sam_resources.py

```diff
@@ -2266,53 +2273,64 @@ def _construct_appsync_api_resources(
         # GraphQLApi will not include logging if and only if the user explicity sets Logging as false boolean.
         # It will for every other value (including true boolean which is essentially same as None).
         if isinstance(model.Logging, bool) and model.Logging is False:
-            return api, None
+            return api, None, auth_connectors
 
         api.LogConfig, cloudwatch_role = self._parse_logging_properties(model)
 
-        return api, cloudwatch_role
+        return api, cloudwatch_role, auth_connectors
 
-    def _parse_auth_properties(self, api: GraphQLApi, auth: aws_serverless_graphqlapi.Auth) -> None:
+    def _parse_and_set_auth_properties(
+        self, api: GraphQLApi, auth: aws_serverless_graphqlapi.Auth
+    ) -> List[Intrinsicable[str]]:
         """
         Parse the Auth properties in a Serverless::GraphQLApi resource.
 
-        This function does not return a value. The GraphQLApi object is modified directly so that we don't
-        return multiple config properties which are mostly None values.
+        Returns: List of Lambda Function arns of Lambda authorizers. If no Lambda authorizer is used, the list is empty.
         """
-        # Default authentication
-        name, auth_dict = self._validate_auth_type_and_config(auth)
+        # Keep all lambda authorizers together to create connectors later
+        lambda_auth_arns: List[Intrinsicable[str]] = []
+
+        # Default authoriser
+        default_auth = aws_serverless_graphqlapi.Authorizer.parse_obj(
+            {k: v for k, v in auth.dict().items() if k != "Additional"}
+        )
+        name, auth_dict = self._validate_and_extract_authorizer_config(default_auth)
         api.AuthenticationType = auth.Type
 
         # This would be much easier and type-safe if accessing the properties in Resource
         # was a dictionary. Currently, top level properties are class attributes, but nested
         # properties are dictionaries...
         # TODO: Access properties in Resource class as dict?
-        if name:
+        if name and auth_dict:
             setattr(api, name, auth_dict)
+            if name == "LambdaAuthorizerConfig":
+                lambda_auth_arns.append(cast(LambdaAuthorizerConfigType, auth_dict)["AuthorizerUri"])
```

Comment:
Will there be more of these `name` types? I notice this is duplicated below.

Reply:
Nope. I didn't want to go in additional refactoring here. But 2 this is it. The only reason to use it because Python3.7 which we still support doesn't support `isinstance(obj, SubTypeOfTypedDict)` :(

Reply:
Would it simplify things if `_validate_and_extract_authorizer_config` returned a `Tuple` of `Optional`s for each type, instead of a single tuple and then branching based off of `name`?

Reply:
Sorry, I'm not following what exactly are you suggesting

Reply:
Currently it returns (name, object), and in caller we do specific logic based on the name. What if instead we return (object, object, object, ...), one for each possible object. And instead of checking the name, we check whether the optional is not `None`.

Reply:
I see. Sure. I'll do it in a separate pr, not to mess up with what I'm doing now